File: app/src/main/java/jp/hazuki/yuzubrowser/webkit/handler/WebSrcImageShareWebHandler.java
package jp.hazuki.yuzubrowser.webkit.handler;

import android.app.Activity;

import java.lang.ref.WeakReference;

import jp.hazuki.yuzubrowser.utils.WebUtils;

public class WebSrcImageShareWebHandler extends WebSrcImageHandler {
    private final WeakReference<Activity> mReference;

    public WebSrcImageShareWebHandler(Activity activity) {
        mReference = new WeakReference<>(activity);
    }

    @Override
    public void handleUrl(String url) {
        Activity activity = mReference.get();
        if (activity != null)
            WebUtils.shareWeb(activity, url, null, null, null);
    }
}

File: app/src/main/java/jp/hazuki/yuzubrowser/webkit/handler/WebSrcImageShareWebHandler.kt
package jp.hazuki.yuzubrowser.webkit.handler

import android.app.Activity
import jp.hazuki.yuzubrowser.utils.WebUtils
import java.lang.ref.WeakReference

class WebSrcImageShareWebHandler(activity: Activity) : WebSrcImageHandler() {
    private val mReference: WeakReference<Activity> = WeakReference(activity)

    override fun handleUrl(url: String) {
        mReference.get()?.let {
            WebUtils.shareWeb(it, url, null)
        }
    }
}
